### ISSUE TITLE
bool to int convertion for expression

### ISSUE THREAD

**Issue by Konstantin8105** [elliotchance/c2go#429]
C:
```c
#include<stdio.h>
struct s { int i; };
int f (void)
{
        struct s *p = 0, *q;
        int j = 0;
        again:
        q = p, p = &((struct s){ j++ });
        if (j < 2) goto again;
        return p == q && q->i == 1;
}

int main(){
        printf("%d\n",f());
        return 0;
}
```
Go:
```go
type s struct {
        i int
}

func f() int {
        var p []s = nil
        var q []s
        var j int = 0
again:
        ;
        q = p
        p = (*[1]s)(unsafe.Pointer(&(s{func() int {
                j += 1
                return j
        }()})))[:]
        if j < 2 {
                goto again
        }
        return map[bool]int{false: 0, true: 1}[p == q && q[0].i == 1]
}
```
Result:
```
○ → go run 5.go
# command-line-arguments
./5.go:201:43: invalid operation: p == q (slice can only be compared to nil)
```

**Comment by Konstantin8105** [comment]
see `EXAMPLE 8 Page 77 . Committee Draft — May 6, 2005  ISO/IEC 9899:TC2`

**Comment by Konstantin8105** [comment]
check after #445

**Comment by elliotchance** [comment]
@Konstantin8105 Was this fixed with #445?

**Comment by Konstantin8105** [comment]
That issue is not solved by #445
